Fix several warnings exposed in MSVS.

null77 opened the pull request:
diagnostic.cpp:
- unreachable code

operand.cpp
- conversion between int and uint32_t
- unreachable code

text.cpp:
- assignment within conditional expression

hex_float.h:
- assignment within conditional expression
- conversion from 'const int' to 'unsigned int'
- unreachable code

validate_id.cpp
- forcing value to bool 'true' or 'false'

validate_types.cpp:
- forcing value to bool 'true' or 'false'

dekimir commented on source/util/hex_float.h:
```diff
@@ -662,12 +659,19 @@ inline uint8_t get_nibble_from_character(int character) {
   const char* upper = "ABCDEF";
-  const char* p = nullptr;
-  if ((p = strchr(dec, character))) {
+  const char* p = p = strchr(dec, character);
```
Does 'p =' really have to be repeated?

dekimir replied:
Interestingly, this line seems to break the gcc build (see "checks have failed" below).

dekimir commented:
BTW, which MSVS version is this?  We seem to have no trouble on 2013 and 2015 with the code as it currently stands.

Also, I would oppose a blanket ban on assignment within conditional expression going forward.

null77 commented:
Updated CL, reverted the assignment within conditional fix. The extra 'p' was a bug, and it's cool that the CI caught that. Updated commit message to show that I'm using VS2015.

dekimir commented:
Rebased and merged to master via 34cb003.
